examples/chip-tool/py_matter_chip_tool_adapter/matter_chip_tool_adapter/encoder.py

```diff
@@ -0,0 +1,326 @@
+        return f'json:{{ {payload} }}'
+
+    def __get_cluster_name(self, request):
+        return self.__get_alias(request.cluster) or self.__format_cluster_name(request.cluster)
+
+    def __get_command_name(self, request):
+        command_name = self.__get_alias(
+            request.cluster, request.command) or self.__format_command_name(request.command)
+
+        # 'readAttribute' is converted to 'read attr-name', 'writeAttribute' is converted to 'write attr-name',
+        # 'readEvent' is converted to 'read event-name', etc.
+        if request.is_attribute:
+            command_specifier = self.__format_command_name(request.attribute)
+            # chip-tool expose writable attribute under the "write" command, but for non-writable
+            # attributes, those appears under the "force-write" command.
+            if command_name == 'write':
+                attribute = self.__specs.get_attribute_by_name(
+                    request.cluster, request.attribute)
+                if attribute and not attribute.is_writable:
+                    command_name = 'force-write'
+        elif request.is_event:
+            command_specifier = self.__format_command_name(request.event)
+        else:
+            command_specifier = None
+
+        return command_name, command_specifier
+
+    def __get_arguments(self, request):
+        arguments = ''
+        arguments = self.__maybe_add_destination(arguments, request)
+        arguments = self.__maybe_add_endpoint(arguments, request)
+        arguments = self.__maybe_add_command_arguments(arguments, request)
+        arguments = self.__maybe_add(
+            arguments, request.min_interval, "min-interval")
+        arguments = self.__maybe_add(
+            arguments, request.max_interval, "max-interval")
+        arguments = self.__maybe_add(arguments, request.timed_interaction_timeout_ms,
+                                     "timedInteractionTimeoutMs")
+        arguments = self.__maybe_add(
+            arguments, request.event_number, "event-min")
+        arguments = self.__maybe_add(
+            arguments, request.busy_wait_ms, "busyWaitForMs")
+        arguments = self.__maybe_add(
+            arguments, request.identity, "commissioner-name")
+        arguments = self.__maybe_add(arguments, request.fabric_filtered,
+                                     "fabric-filtered")
+
+        return arguments
+
+    def __maybe_add_destination(self, rv, request):
+        if not self._has_destination_supports(request):
+            return rv
+
+        destination_argument_name = 'destination-id'
+        destination_argument_value = None
+
+        if request.group_id:
+            destination_argument_value = hex(
+                0xffffffffffff0000 | int(request.group_id))
+        elif request.node_id:
+            destination_argument_value = hex(request.node_id)
+        else:
+            destination_argument_value = None
+
+        if rv:
+            rv += ', '
```

Comment:
That is likely. I have opened https://github.com/project-chip/connectedhomeip/issues/25307 to do that in a followup. It will change a lot of places and I'm afraid to miss an obvious case and block this PR on that.